apiserver/facades/controller/instancepoller/instancepoller.go

```diff
@@ -138,11 +138,19 @@ func (a *InstancePollerAPI) ProviderAddresses(args params.Entities) (params.Mach
 	}
 	for i, arg := range args.Entities {
 		machine, err := a.getOneMachine(arg.Tag, canAccess)
-		if err == nil {
-			addrs := machine.ProviderAddresses()
-			result.Results[i].Addresses = params.FromNetworkAddresses(addrs...)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(err)
+			continue
 		}
-		result.Results[i].Error = common.ServerError(err)
+
+		addrs, err := machine.ProviderAddresses().ToProviderAddresses(a.st)
```

Comment:
Outgoing addresses are converted via lookup back to `ProviderAddresses`. No `SpaceAddress` should ever be observed client-side.